### ISSUE TITLE
Add multihop routes to SwapMsgs

### ISSUE THREAD

**Issue by sunnya97** [osmosis-labs/osmosis#33]
If people want to do a trade between two assets that aren’t in the same pool, we need to construct a multihop swap.

To do this, we should switch the `PoolId` in a SwapMsg to into a `Route` which is an ordered list of PoolIds.

So for example:

```
type MsgSwapExactAmountIn struct {
	Sender            string
	PoolId            uint64
	TokenIn           types.Coin
	TokenOutDenom     string
	TokenOutMinAmount github_com_cosmos_cosmos_sdk_types.Int
	MaxSpotPrice      github_com_cosmos_cosmos_sdk_types.Dec
}
``` 
would become
```
type MsgSwapExactAmountIn struct {
	Sender            string
	Route             []PoolId (int64)
	TokenIn           types.Coin
	TokenOutDenom     string
	TokenOutMinAmount github_com_cosmos_cosmos_sdk_types.Int
	MaxSpotPrice      github_com_cosmos_cosmos_sdk_types.Dec
}
```

**Comment by Thunnini** [comment]
I don’t think `Route []PoolID (int64)` would be enough. If there is an intermediary pool, it may be necessary to know the intermediary token’s denom. So perhaps Route [](PoolId, OutDenom) (int64, string) may be a better format, and remove `TokenOutDenom`.

Also, for cases where the transaction is routed through an intermediary pool, how do you think the swap fee should be handled? Should routed transactions pay the same amount of swap fees? or should there be some form of swap fee discounts for router transactions?

**Comment by Thunnini** [comment]
What do you think about adding routerSwapFee that sets a separate swap fee for intermediary router transaction as a pool parameter which can be adjusted through the pool governance?

**Comment by Thunnini** [comment]
Anyway, I can implement this one.

**Comment by sunnya97** [comment]
> I don’t think Route []PoolID (int64) would be enough. If there is an intermediary pool, it may be necessary to know the intermediary token’s denom. So perhaps Route [](PoolId, OutDenom) (int64, string) may be a better format, and remove TokenOutDenom.

Oh good point...  Yeah, was thinking only in terms of two token pools.  

> Also, for cases where the transaction is routed through an intermediary pool, how do you think the swap fee should be handled? 

I think the swap fee should be paid to each pool normally.  What's the reason it should be discounted?  The way I see it, you're still using the services of the pool, and causing the same effects on the pool as if you were just doing a single swap.

**Comment by sunnya97** [comment]
closed by #35
